sui/transfer-object.js

```diff
@@ -0,0 +1,93 @@
+const { TransactionBlock } = require('@mysten/sui.js/transactions');
+const { Command, Option } = require('commander');
+const { printInfo, validateParameters } = require('../evm/utils');
+const { addExtendedOptions } = require('./cli-utils');
+const { getWallet, printWalletInfo } = require('./sign-utils');
+const { loadSuiConfig } = require('./utils');
+
+async function processCommand(chain, options) {
+    const [keypair, client] = getWallet(chain, options);
+    await printWalletInfo(keypair, client, chain, options);
+    const recipient = options.recipient;
+    validateParameters({
+        isKeccak256Hash: { recipient },
+    });
+
+    let objectId;
+
+    if (options.objectId) {
+        objectId = options.objectId;
+        validateParameters({
+            isKeccak256Hash: { objectId },
+        });
+    } else if (options.contractName && options.objectName) {
+        const { contractName, objectName } = options;
+
+        validateParameters({
+            isString: { contractName, objectName },
+        });
+        const contractsData = chain.contracts;
+
+        if (!contractsData) {
+            throw new Error(`contract data not found`);
+        }
+
+        const contractObject = contractsData[`${contractName}`];
+
+        if (!contractObject) {
+            throw new Error(`contract name [${contractName}] not found`);
+        }
+
+        const objectsData = contractObject.objects;
+
+        if (!objectsData) {
+            throw new Error(`contract objects not found`);
+        }
+
+        objectId = objectsData[`${objectName}`];
+
+        if (!objectId) {
+            throw new Error(`object name [${objectName}] not found`);
+        }
```

Comment:
Very verbose, use the optional chaining operator to get the objectId directly from chain. E.g. `chain.contracts?.[contractName]...` and validate the objectId at the end outside the if blocks generically. See our usage of ?. operator in other scripts.

Reply:
no, it was just for readability of the intermediate objects, since the line was quite long and complex. `?` simplifies validation checks